feat(app): Add stop function to to turn off the camera

thaiat opened the pull request:
see #7

amw commented on dist/jpeg_camera.js:
```diff
@@ -304,6 +304,7 @@
         }
         get_user_media_options = {
           video: {
+            facingMode: "environment",
```
Looks like this isn't used anywhere and can be removed.

amw commented:
Thank you for your pull request. A couple questions:

1. What's the expected usage here? You call stop, the stream goes black and then you manually remove the camera element from the DOM? Could please add note to README?
2. You've added a public method "stop" only to the html5 engine. I support both html5 and flash. The way I do it is I add public methods (with appropriate documentation comment) to JpegCamera class and these methods call internal methods like _engine_do_something that need to be implemented by both html5 and flash engine.

If you can't/don't want to edit the flash part then add empty no-op method for flash engine and I might get around to implementing it if it's necessary (it might not need be). Structure your calls the way I described and add documentation for your method.

thaiat commented:
I have added the stop method on jpeg_camera.coffee so it should exist in any subclass including jpeg_camera_flash.
Unfortunaltely i was not able to run grunt dist (it fails) and so the dist folder just reflect a `grunt js`

Lastly i have updated the README and switch to using the formatting recommended syntax for code so you get nicer color when you read it.

Let me know what you think

amw commented:
Thanks for your commits. I am on vacation until July 25th so please give me a couple weeks to review them.

amw commented:
I'm sorry I haven't responded to this pull request earlier. I've seen I couldn't merge this, but didn't take time to articulate why.

I can't accept or even properly discuss this PR in this form because it contains too many unrelated changes. I see a few different PRs here:
- documentation changed from indented to block style (would be willing to merge a PR for this)
- change how `mirror` affects display on screen (would reject this)
- new syntax for getUserMedia resolution (needs discussion on a separate PR)
- canvasToBlob changes (already updated in response to #12)
- the `stop` functionality discussed in this PR (which still doesn't follow coding style I've suggested in earlier comment, e.g. `_engine_do_something` method)

If you are still interested in merging changes from this PR please submit new pull requests from separate topic branches for each unrelated change. You can read about topic branches:
https://help.github.com/articles/about-pull-requests/
https://github.com/dchelimsky/rspec/wiki/Topic-Branches
